src/views/figure_editable.ts

```diff
@@ -50,11 +60,160 @@ export class FigureEditableView extends FigureView {
   public reactTools: HTMLDivElement
   positionMenuWrapper: HTMLDivElement
   figurePosition: string
+  private isDragging = false
+  private dragHandle: HTMLDivElement | undefined
 
   public initialise() {
     this.upload = this.upload.bind(this)
     this.createDOM()
     this.updateContents()
+    if (this.props.getCapabilities().editArticle) {
+      this.setupDragAndDrop()
+    }
+  }
+
+  private setupDragAndDrop() {
+    this.container.draggable = false
+
+    // Drag events for container
+    this.container.addEventListener('dragstart', (e) => {
+      if (!e.dataTransfer) {
+        return
+      }
+      this.isDragging = true
+      const pos = this.getPos()
+      const figureId = this.node.attrs.id
+      figureDragState.setActiveDrag(figureId, pos)
+      e.dataTransfer.setData('text/plain', figureId)
+      this.container.classList.add('dragging')
+      // Add drag-active to siblings only
+      const parent = this.container.parentElement
+      if (parent) {
+        Array.from(parent.children).forEach((el) => {
+          if (el !== this.container && el.classList.contains('figure')) {
+            el.classList.add('drag-active')
+          }
+        })
+      }
+    })
+
+    // Drag events for drag handle (if present)
+    if (this.dragHandle) {
+      this.dragHandle.addEventListener('dragstart', (e) => {
+        if (!e.dataTransfer) {
+          return
+        }
+        this.isDragging = true
+        const pos = this.getPos()
+        const figureId = this.node.attrs.id
+        figureDragState.setActiveDrag(figureId, pos)
+        e.dataTransfer.setData('text/plain', figureId)
+        this.container.classList.add('dragging')
+        // Add drag-active to siblings only
+        const parent = this.container.parentElement
+        if (parent) {
+          Array.from(parent.children).forEach((el) => {
```

Comment:
It would be more reliable and simpler to use parentElement.querySelector to find figures. Ideally though, we should set decorations either on parent element or on the figures without directly modifying their classList.